sdk/core/System.ClientModel/src/Internal/TaskExtensions.cs

```diff
@@ -0,0 +1,309 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ClientModel.Internal;
+
+internal static class TaskExtensions
```

Comment:
This type is copied from Azure.Core. Tracking this technical debt with https://github.com/Azure/azure-sdk-for-net/issues/41035